pkg/fanal/artifact/remote/git_test.go

```diff
@@ -177,3 +177,132 @@ func Test_newURL(t *testing.T) {
 		})
 	}
 }
+
+func Test_branch(t *testing.T) {
+	type args struct {
+		rawurl      string
+		c           cache.ArtifactCache
+		FetchBranch string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr string
+	}{
+		{
+			name: "happy branch",
+			args: args{
+				rawurl:      "https://github.com/aquasecurity/trivy-plugin-aqua",
```

Comment:
We don't depend on the Internet in unit tests. Please think about the use of a dummy server.
https://github.com/aquasecurity/trivy/blob/11073799102054f2a8340a2c1e8e126501c9d6f8/pkg/fanal/artifact/remote/git_test.go#L17-L30

Reply:
We have a bare git repository in `testdata`.

```
$ cd testdata
$ git clone -l ./test.git ./test
$ cd test
$ git tag v1.0.0
$ git push
```
I hope it works.

Reply:
Thank you for this, was able to do it (I hope) correctly!